### ISSUE TITLE
rds.dataSourceEngineVersionRead panic

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-aws#3752]
Panic in rds.dataSourceEngineVersionRead reported by customer.

```
         panic: interface conversion: interface {} is nil, not map[string]interface {}
         
         goroutine 44 [running]:
         
         github.com/hashicorp/go-cty/cty.Value.GetAttr({{{0x1380f628?, 0xc002c1c758?}}, {0x0?, 0x0?}}, {0x11c8040c, 0x6})
         
         	/home/runner/go/pkg/mod/github.com/hashicorp/go-cty@v1.4.1-0.20200723130312-85980079f637/cty/value_ops.go:711 +0x2de
         
         github.com/hashicorp/terraform-provider-aws/internal/service/rds.criteriaSet(0xc003896800?, {0xc003896ec0?, 0x9, 0x10?})
         
         	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/rds/engine_version_data_source.go:481 +0x9b
         
         github.com/hashicorp/terraform-provider-aws/internal/service/rds.dataSourceEngineVersionRead({0x1380f1c8, 0xc00383e660}, 0x0?, {0x11c1f880?, 0xc00219d500?})
         
         	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/service/rds/engine_version_data_source.go:227 +0x645
         
         github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedDataSource).Read.interceptedHandler[...].func7(0x0?, {0x11c1f880?, 0xc00219d500?})
         
         	/home/runner/work/pulumi-aws/pulumi-aws/upstream/internal/provider/intercept.go:112 +0x283
         
         github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1380f1c8?, {0x1380f1c8?, 0xc00380fbc0?}, 0xd?, {0x11c1f880?, 0xc00219d500?})
         
         	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20240229143312-4f60ee4e2975/helper/schema/resource.go:795 +0x7a
         
         github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc001d9c0e0, {0x1380f1c8, 0xc00380fbc0}, 0xc003902280, {0x11c1f880, 0xc00219d500})
         
         	/home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20240229143312-4f60ee4e2975/helper/schema/resource.go:1020 +0x149
         
         github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.ReadDataApply({0xc000912600?, {0xc0023a1580?, 0xe726116?, 0xc0036bf680?}}, {0x1380f1c8, 0xc00380fbc0}, {0x11d36431, 0x16}, {0x13821640, 0xc003902280})
         
         	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.77.1-0.20240311191722-54802d5b003c/pkg/tfshim/sdk-v2/provider.go:177 +0xcb
         
         github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Invoke(0xc00225a000, {0x1380f1c8?, 0xc00380f830?}, 0xc0038da4c0)
         
         	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.77.1-0.20240311191722-54802d5b003c/pkg/tfbridge/provider.go:1434 +0xb46
         
         github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Invoke(0xc0029f4e40, {0x1380f1c8, 0xc00380f830}, 0xc0038da4c0)
         
         	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.8/muxer.go:342 +0x10e
         
         github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Invoke_Handler.func1({0x1380f1c8, 0xc00380f830}, {0x10e53260?, 0xc0038da4c0})
         
         	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.108.1/proto/go/provider_grpc.pb.go:500 +0x75
         
         github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x1380f1c8, 0xc0038a2f60}, {0x10e53260, 0xc0038da4c0}, 0xc003805b60, 0xc002b70108)
         
         	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x3d0
         
         github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Invoke_Handler({0x11a51c40?, 0xc0029f4e40}, {0x1380f1c8, 0xc0038a2f60}, 0xc0038a4300, 0xc002cbe900)
         
         	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.108.1/proto/go/provider_grpc.pb.go:502 +0x135
         
         google.golang.org/grpc.(*Server).processUnaryRPC(0xc002cf8000, {0x1380f1c8, 0xc0038a2ed0}, {0x1384c420, 0xc001a376c0}, 0xc0038acc60, 0xc002a66600, 0x1d8d7900, 0x0)
         
         	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.0/server.go:1383 +0xe03
         
         google.golang.org/grpc.(*Server).handleStream(0xc002cf8000, {0x1384c420, 0xc001a376c0}, 0xc0038acc60)
         
         	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.0/server.go:1794 +0x100c
         
         google.golang.org/grpc.(*Server).serveStreams.func2.1()
         
         	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.0/server.go:1027 +0x8b
         
         created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
         
         	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.0/server.go:1038 +0x135

```

First version of provide where this happens: 6.27.0

**Comment by mikhailshilkov** [comment]
The panic happens in https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/rds/engine_version_data_source.go#L481 when we try to access a nil `RawConfig`. This code block has been recently introduced in https://github.com/hashicorp/terraform-provider-aws/commit/f8ab2903c9b0d9b30c53ad14e968bd54e2b93e30

It sounds similar to https://github.com/pulumi/pulumi-terraform-bridge/pull/1332, although not in Diff but in data source Read.

**Comment by t0yv0** [comment]
Thanks Mikhail. Yes this makes sense, we don't have RawConfig fidelity yet but pass approximate values, and it looks like in Read we're not passing anything at all (nil)? Thanks for the repro.

**Comment by mikhailshilkov** [comment]
@corymhall Is this now fixed with #3757? Can we close?

**Comment by t0yv0** [comment]
This is fixed in https://github.com/pulumi/pulumi-aws/releases/tag/v6.28.2  - @corymhall is out this week. I'll close.
